### DISCUSSION TITLE
Offload Build and Deploy Process to GitHub Actions

### DISCUSSION THREAD

**Discussion by SudharakaP** [opencollective/opencollective#4399] [Ideas]
Something I was thinking while doing a deployment. 🤔 

Currently our deployment process is as follows; https://github.com/opencollective/opencollective-frontend/blob/main/docs/deployment.md

So each developer will do this in their local machine. It actually includes some additional steps that I think we all do. 

- Make sure we are on main branch and no local changes. 🐨 
- Make sure there's no local environment variables defined in `.env`. 🐨 
- While doing the deploy not change branches or do local changes; at least I am scared to do so. So wait until it's done. 😄 

Given this I was wondering if there's an alternative. Maybe we could write a GitHub action to do this so that when we create a tag, it will trigger the build. I am just wondering if this is something worthwhile. The point being to avoid anything that could potentially go wrong in the deployment process. Also this way while the deployment is triggered you could use your local to do anything else without fear. 😄 

@opencollective/core-developers

**Comment by Betree** [comment]
> Make sure we are on main branch and no local changes. koala

The deploy script is basically just a `git push production main,` so it's not a big deal if you're not on `main,` as long as it's up to date.

> Make sure there's no local environment variables defined in .env. koala

`.env` is not versioned, so that should not be a problem. Or am I missing something?

> While doing the deploy not change branches or do local changes; at least I am scared to do so. So wait until it's done. smile

That is normally not a problem either. Once the initial push is done, all you see in the terminal is the build log - but your local env is not involved anymore.

> Given this I was wondering if there's an alternative. Maybe we could write a GitHub action to do this so that when we create a tag, it will trigger the build. I am just wondering if this is something worthwhile. The point being to avoid anything that could potentially go wrong in the deployment process. Also this way while the deployment is triggered you could use your local to do anything else without fear. smile

We could eventually change our deployment process; feel free to propose something if you think that can improve the way we work. I'm personally attached to how we do things today; it gives us a lot of control over the process while being simple to hack and debug (it's only git branches).

I'd be worried about putting deploy secrets in our Github actions, given that many contributors can trigger them.

**Reply by SudharakaP** [reply]
>The deploy script is basically just a git push production main, so it's not a big deal if you're not on main, as long as it's up to date.

>`.env` is not versioned, so that should not be a problem. Or am I missing something?

>That is normally not a problem either. Once the initial push is done, all you see in the terminal is the build log - but your local env is not involved anymore.

@Betree : Ah, I now that I think about it you are right. I don't see any need to change this if these points aren't a problem. Somehow I thought they were, especially the third one; my bad. 😄 😄
